### ISSUE TITLE
Duke Sucellus demonbane reduction seems wrong

### ISSUE THREAD

**Issue by SuperNerdEric** [weirdgloop/osrs-dps-calc#108]
### What went wrong?

According this guys account he hit a 51 against Duke Sucellus with Arclight with 75 strength bonus, 118 strength, and piety.
https://twitter.com/Josh_H__/status/1687938409879257092

However with the same setup the calc is saying max hit is 39.
https://dps.osrs.wiki/?id=e0585acb-967b-4bb3-8b25-83fb057981b8
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/7608429/289d40b2-e0a0-43db-bf4d-6ceb4bb5a01d)

The Bitterkoekje sheet also says it should hit a 51.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/7608429/339a99bb-ea72-4474-9223-7f54e89ba0d8)

I think the current way the reduction (probably for both damage and attack roll) is being applied incorrectly. Also note that Mod Ash says that the effective boost from Arclight against the Duke is 50%, not 49% (70% * 70%).
https://twitter.com/JagexAsh/status/1688465428912242688

```
    if (this.wearing('Arclight') && mattrs.includes(MonsterAttribute.DEMON)) {
      maxHit = Math.trunc(maxHit * 17 / 10);
      if (this.monster.name === 'Duke Sucellus') {
        maxHit = Math.trunc(maxHit * 7 / 10);
      }
    }
```

### What browsers are you seeing the problem on?

Firefox

### What device(s) are you seeing the problem on?

Windows

**Comment by LlemonDuck** [comment]
This is confusing me a bit, since it doesn't seem doable in a sensible way using integer math only. I've been trying a bunch of methods but none come out to a max hit of 51 under that setup. It doesn't really make sense that it would round up, we don't see that kind of behaviour anywhere else.

I'm thinking it must be special-cased as 50%, and maybe the other Duke demonbane values have been as well, but some more research would be great. I hesitate to special-case only this and would prefer to discover the underlying formula before making changes.

Maybe I'll farm some Duke with varying setups and track my max hits this week, and I would welcome if anyone else has data on this too.

**Comment by SuperNerdEric** [comment]
Yeah I'm assuming it's just applying 50% directly instead of 70% then some sort of reduction. But then we would need to know the effective bonus of every single Demonbane application.

FWIW the only case that the Bitterkoekje sheet has bothered to implement is Arclight as 50%
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/7608429/4e8f08b6-3817-40de-a123-e5610cccdac8)

**Comment by crab-pancake** [comment]
i believe the multiplier may have been changed since that tweet was made. according to stillremains/alex (recently grinded for duke wr) max hit is 50 rather than 51.

**Comment by LlemonDuck** [comment]
@crab-pancake do you know what gear stillremains was using when they got a max hit of 50? I was able to reproduce with this setup, but get a different result when using Ultor.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/1868974/2766b4de-1ed3-45d3-928c-abe4023dc864)

**Comment by LlemonDuck** [comment]
I've landed on an approach in b8c4c0fbeb566d773bb435466d043b5a15d7164d that seems correct to me, but I haven't been able to verify on extra setups. I could use any and all data on setups with their max hits, if anyone has any to share.

Keeping this open as research for more verification, but no longer bug.

**Comment by LlemonDuck** [comment]
fixed in 6a25972adb02d806cc97580a582efd0aa0621ef3 btw
